examples/pascal_voc_jaccard_loss.py

```diff
@@ -0,0 +1,129 @@
+'''
+Trains a DenseNet-40-12 model on a reduced Pascal VOC segmentation dataset.
+
+Gets a 70% accuracy score after 100 epochs.
+'''
+from __future__ import absolute_import
+from __future__ import print_function
+from __future__ import division
+
+import numpy as np
+import os
+
+from keras import backend as K
+from keras.utils.np_utils import to_categorical
+from keras.callbacks import ModelCheckpoint, ReduceLROnPlateau, EarlyStopping
+from keras.optimizers import Adam
+from keras.preprocessing.image import ImageDataGenerator
+from keras.utils import np_utils
+from keras.losses import categorical_crossentropy
+from keras_contrib.losses.jaccard import jaccard_distance
+from keras_contrib.datasets import pascal_voc
+from keras_contrib.applications.densenet import DenseNetFCN
+
+batch_size = 64
+nb_classes = 21
+epochs = 100
+
+img_rows, img_cols = 32, 32
+img_channels = 3
+
+# Parameters for the DenseNet model builder
+img_dim = (img_channels, img_rows,
+           img_cols) if K.image_data_format() == 'channels_first' else (
+               img_rows, img_cols, img_channels)
+depth = 40
+nb_dense_block = 3
+growth_rate = 12
+nb_filter = 16
+dropout_rate = 0.0  # 0.0 for data augmentation
+
+conf = pascal_voc.voc_config()
+pascal_folder = os.path.join(conf['pascal_berkeley_root'], 'dataset')
+
+# download dataset.. about an hour but you only need to do it once
+if not os.path.isdir(pascal_folder):
+    pascal_voc.data_pascal_voc.run()
+
+# load the data using dual imagedata generators
+# we create two instances with the same arguments
+data_gen_args = dict()
+image_datagen = ImageDataGenerator(**data_gen_args)
+mask_datagen = ImageDataGenerator(**data_gen_args)
+
+# Provide the same seed and keyword arguments to the fit and flow methods
+seed = 1
+image_generator = image_datagen.flow_from_directory(
+    pascal_folder,
+    target_size=(img_rows, img_cols),
+    class_mode=None,
+    classes=["img"],
+    seed=seed)
+mask_generator = image_datagen.flow_from_directory(
+    pascal_folder,
+    target_size=(img_rows, img_cols),
```

Comment:
resizing the labels this way will change the label values, and training won't work correctly. Consider merging SegDataGenerator from https://github.com/aurora95/Keras-FCN/blob/master/utils/SegDataGenerator.py#L291. A usage example is available at https://github.com/aurora95/Keras-FCN/blob/master/train.py#L117

Reply:
Oh shit, you're right. Thanks for catching that. I'll do that or a basic decimation (I only have 2GB gpu).